src/plugins/explore/public/components/query_panel/components/footer/index.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { EuiButtonEmpty, EuiFlexGroup, EuiFlexItem, EuiText } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { ShowFieldToggle } from './show_field';
+import { SaveQueryButton } from './save_query';
+import { Actions } from './actions';
+import { DateTimeRangePicker } from './date_time_selector';
+import { RunQueryButton } from './run_query';
+import { ShowInputType } from './show_input_type';
+import { LanguageType } from '../../types';
+import { QueryError } from './query_error';
+import { ResultStatus } from '../../types';
+
+import './index.scss';
+
+const RECENT_QUERIES_LABEL = i18n.translate('explore.queryPanel.recentQueryLabel', {
+  defaultMessage: 'Recent Queries',
+});
+
+interface QueryPanelFooterProps {
+  languageType: LanguageType;
+  onRunClick: () => void;
+  onRecentClick: () => void;
+  isDualEditor: boolean;
+  isLoading: boolean;
+  noInput: boolean;
+  lineCount: number | undefined;
+}
+
+export const QueryPanelFooter: React.FC<QueryPanelFooterProps> = ({
+  languageType,
+  onRunClick,
+  onRecentClick,
+  isDualEditor,
+  isLoading,
+  noInput,
+  lineCount,
+}) => {
+  const showLineCount =
+    typeof lineCount === 'number' &&
+    lineCount > 0 &&
+    (languageType !== LanguageType.Natural || isDualEditor); // Show line number only for ppl or dual
+  return (
+    <div className="queryPanel__footer">
+      <EuiFlexGroup
+        justifyContent="spaceBetween"
+        alignItems="center"
+        gutterSize="xs"
+        responsive={true}
+        wrap
+      >
+        {/* Left Section */}
+        <EuiFlexItem grow={1} className="queryPanel__footer__minWidth0">
+          <EuiFlexGroup alignItems="center" gutterSize="xs" responsive={true} wrap>
+            <EuiFlexItem grow={false}>
+              <ShowFieldToggle
+                isEnabled={true}
+                onToggle={(enabled) => {
+                  // Todo: Dispatch query action to update toggle value which can be used by field sidebar
+                }}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <div className="queryPanel__footer__verticalSeparator" />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty
+                onClick={onRecentClick}
+                iconType="clock"
+                className="queryPanel__footer__recentQueriesButton"
+                data-test-subj="queryPanelFooterRecentQueriesButton"
+              >
+                {RECENT_QUERIES_LABEL}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <div className="queryPanel__footer__verticalSeparator" />
```

Comment:
you have responsive set to true on the flex group with vertical separator, they won't work well in mobile view where items are vertical. maybe remove responsive or use other ways to add separator?